### PR TITLE
CLOUDSTACK-9641 In KVM SSVM and CPVM may use the old cmdline data, if…

### PR DESCRIPTION
**Pull request by bvbharatk**
If we fail to fetch the new cmdline, we do not fail ssvm start. we end up using the old cmdline.  So to fix this we are moving the oldcmdline to backup and if we do not find the new one, we fail and restore the old cmdline back.

### CODE REVIEW

**Comment by cloudmonger** [comment]
### ACS CI BVT Run
 **Sumarry:**
 Build Number 716
 Hypervisor xenserver
 NetworkType Advanced
 Passed=112
 Failed=0
 Skipped=12

_Link to logs Folder (search by build_no):_ https://www.dropbox.com/sh/r2si930m8xxzavs/AAAzNrnoF1fC3auFrvsKo_8-a?dl=0

**Failed tests:**

**Skipped tests:**
test_vm_nic_adapter_vmxnet3
test_01_verify_libvirt
test_02_verify_libvirt_after_restart
test_03_verify_libvirt_attach_disk
test_04_verify_guest_lspci
test_05_change_vm_ostype_restart
test_06_verify_guest_lspci_again
test_static_role_account_acls
test_11_ss_nfs_version_on_ssvm
test_nested_virtualization_vmware
test_3d_gpu_support
test_deploy_vgpu_enabled_vm

**Passed test suits:**
test_deploy_vm_with_userdata.py
test_affinity_groups_projects.py
test_portable_publicip.py
test_vm_snapshots.py
test_over_provisioning.py
test_global_settings.py
test_scale_vm.py
test_service_offerings.py
test_routers_iptables_default_policy.py
test_loadbalance.py
test_routers.py
test_reset_vm_on_reboot.py
test_deploy_vms_with_varied_deploymentplanners.py
test_network.py
test_router_dns.py
test_non_contigiousvlan.py
test_login.py
test_deploy_vm_iso.py
test_list_ids_parameter.py
test_public_ip_range.py
test_multipleips_per_nic.py
test_metrics_api.py
test_regions.py
test_affinity_groups.py
test_network_acl.py
test_pvlan.py
test_volumes.py
test_nic.py
test_deploy_vm_root_resize.py
test_resource_detail.py
test_secondary_storage.py
test_vm_life_cycle.py
test_routers_network_ops.py
test_disk_offerings.py

File: plugins/hypervisors/kvm/src/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtStopCommandWrapper.java [hunk 1]
```diff
@@ -59,8 +64,21 @@ public Answer execute(final StopCommand command, final LibvirtComputingResource
                 s_logger.debug("Failed to get vm status in case of checkboforecleanup is true", e);
             }
         }
-
+        File pemFile=new File("/root/.ssh/id_rsa.cloud");
```
Comment (kishankavala) [inline plugins/hypervisors/kvm/src/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtStopCommandWrapper.java]: libvirtComputingResource.SSHPRVKEYPATH can be used instead

File: engine/orchestration/src/com/cloud/vm/VirtualMachineManagerImpl.java [hunk 2]
```diff
@@ -2669,12 +2673,25 @@ private void orchestrateReboot(final String vmUuid, final Map<VirtualMachineProf
...
+        cmd.setControlIp(getControlNicIpForVM(vm));
+        return cmd;
     }
 
+    private String getControlNicIpForVM(VirtualMachine vm){
+        if (vm.getType()== VirtualMachine.Type.ConsoleProxy || vm.getType()== VirtualMachine.Type.SecondaryStorageVm){
+            NicVO nic = _nicsDao.getControlNicForVM(vm.getId());
+            return nic.getIPv4Address();
+        }
+        else if(vm.getType()== VirtualMachine.Type.DomainRouter) return vm.getPrivateIpAddress();
```
Comment (jayapalu) [inline engine/orchestration/src/com/cloud/vm/VirtualMachineManagerImpl.java]: except If coding convention ... code LGTM

**Comment by bvbharatk** [comment]
test_ssvm.py was also run by CI pasting the results xml file here.
[test_ssvm.txt](https://github.com/apache/cloudstack/files/1013930/test_ssvm.txt)

File: plugins/hypervisors/kvm/src/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtStopCommandWrapper.java [hunk 3]
```diff
@@ -83,7 +101,18 @@ public Answer execute(final StopCommand command, final LibvirtComputingResource
 
             return new StopAnswer(command, result, true);
         } catch (final LibvirtException e) {
+            s_logger.debug("unable to stop VM:"+vmName+" due to"+e.getMessage());
+            try{
+                if(vmName.contains("s-") || vmName.contains("v-"))
```
Comment (sureshanaparti) [inline plugins/hypervisors/kvm/src/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtStopCommandWrapper.java]: `startsWith()` here

**Comment by bvbharatk** [comment]
Made the suggested changes.

**Comment by sureshanaparti** [comment]
Code LGTM

**Pull request merged**